CMakeLists.txt

```diff
@@ -0,0 +1,191 @@
+cmake_minimum_required(VERSION 2.6)
+
+project(UriParser C)
+
+if("${CMAKE_BUILD_TYPE}" STREQUAL "")
+  set(CMAKE_BUILD_TYPE Release)

```

Comment:
That part would maybe be nice to limit to non-Unix (or Windows) to support passing `CFLAGS` and friends without having CMake append to it. Not sure yet.

Need to continue review later.